rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -81,3 +83,28 @@ TEST_F(CompressionHelperFixture, zstd_compress_file_uri)
   EXPECT_GT(compressed_file_size, 0u);
   EXPECT_TRUE(rosbag2_storage::FilesystemHelper::file_exists(compressed_uri));
 }
+
+TEST_F(CompressionHelperFixture, zstd_decompress_file_uri)
+{
+  const auto uri = rosbag2_storage::FilesystemHelper::concat({temporary_dir_path_, "file1.txt"});
+  create_garbage_file(uri);
+  const auto initial_file_size = rosbag2_storage::FilesystemHelper::get_file_size(uri);
+
+  auto zstd_compressor = rosbag2_compression::ZstdCompressor();
+  const auto compressed_uri = zstd_compressor.compress_uri(uri);
+
+  ASSERT_EQ(0, std::remove(uri.c_str()));  // The test is invalid if the initial file is not deleted
+
+  auto zstd_decompressor = rosbag2_compression::ZstdDecompressor();
+  const auto decompressed_uri = zstd_decompressor.decompress_uri(compressed_uri);
+
+  const auto expected_decompressed_uri = uri;
+  const auto decompressed_file_size =
+    rosbag2_storage::FilesystemHelper::get_file_size(decompressed_uri);
+
+  EXPECT_NE(compressed_uri, uri);
+  EXPECT_NE(decompressed_uri, compressed_uri);
+  EXPECT_EQ(uri, expected_decompressed_uri);
```

Comment:
Can you add a test comparing the contents of the files at `uri` and `decompressed_uri`?

Reply:
Would this test just verify that the file contents are not the same?

Reply:
You take a file, compress it, then decompress it, and it will end up different? Why is that?

Reply:
Oh I misread the your comment. I'll see what I can do for that.

Reply:
Added unit test that checks decompressed file contents against initial.